src/bigint-encoder.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Encode BigInt value from arbitrary provided arguments
+ *
+ * @param {Array<Number|BigInt|String>} parts - Slices to encode, where the
+ *    earlier elements represent higher bits in the final integer
+ * @param {64|128|256} size - Number of bits in the target integer type
+ * @param {Boolean} unsigned - Whether it's an unsigned integer
+ *
+ * @returns {BigInt}
+ */
+export function encodeBigIntFromBits(parts, size, unsigned) {
+  let result = 0n;
+  // check arguments length
+  if (parts.length && parts[0] instanceof Array) {
+    parts = parts[0];
+  }
```

Comment:
should a guard check happen on `size=`64|128|256`, default or throw if not?

Reply:
done!